CMakeLists.txt

```diff
@@ -18,6 +18,29 @@ option(DOWNLOAD_STB_IMAGE "Download stb_image for image loading." ON)
 option(BUILD_GO_SHLIB "Build Go shared library." OFF)
 option(BUILD_DOCS "Build doxygen documentation (if doxygen is available)." ON)
 
+if(CMAKE_CROSSCOMPILING)
+  set(BOARD_NAME "" CACHE STRING "Enter the board name to compile mlpack for.")
+  set(GFORTRAN_LIBRARY "" CACHE STRING "Path for gfortran library.")
+  set(GCC_VERSION ""  CACHE STRING "")
+endif()
+
+if (CMAKE_CROSSCOMPILING AND (BOARD_NAME STREQUAL ""))
+  message(FATAL_ERROR "Please enter the board name")
+elseif(CMAKE_CROSSCOMPILING AND (BOARD_NAME STRGREATER ""))
+  include(board/flags-config.cmake)
+  flags(${BOARD_NAME})
+
+  set(OPENBLAS_LIBRARIES "${CMAKE_BINARY_DIR}/../OpenBLAS-0.3.13/libopenblas.a" CACHE STRING "OpenBLAS PATH")
+  set(CEREAL_INCLUDE_DIR "${CMAKE_BINARY_DIR}/../cereal-1.3.0/include" CACHE STRING "Cereal header PATH")
+  set(Boost_INCLUDE_DIRS "${CMAKE_BINARY_DIR}/../boost_1_75_0" CACHE STRING "Boost header PATH")
+  set(ARMADILLO_INCLUDE_DIR "${CMAKE_BINARY_DIR}/../armadillo-8.400.0/include" CACHE STRING "Armadillo header PATH")
```

Comment:
Can we still use the CMake library search functionality here?  It seems like all we really need to do here is get GFORTRAN_LIBRARY set right and OPENBLAS_LIBRARIES set to a statically linked version of OpenBLAS for the target architecture.

Reply:
I probably miss something, but why do we list specific versions here?

Reply:
@rcurtin  I know, I totally agree with you, using the already installed version will make our life easier, but unfortunately we can not use the system header, in fact we can not use anything from the system root file when cross compiling.
The reason for this is related to C/C++ headers and anything related to (binaries, libraries (libpthread). etc.). Knowing that C++ headers for aaarch64 are different from the x86, therefore we need to tell the CMake where is the new root for C++ libraries and for relative dependencies. 
I know, this is the main problem when cross compiling, we need to keep two root file systems, one for the host and one for the target. Usually cross-compiling platforms such as `buildroot` download all the dependencies for host and target and respectively compile and cross-compile them one by one in order to create a full Linux system image.
What is good in our case, it that we have to download only-headers libraries, with no requirement for cross compilation. 
The only cross-compiling we need to do is for OpenBLAS.
Let me know what do you think, if you have a better idea let me know :+1:

The second main issue here is related to the toolchain, the user has  to install the toolchain, and I suppose also that is if the user want to cross-compile, then the user has a relative good knowledge about toolchains. 

The best solution I can see for a cross compilation issue can be similar to what build `buildroot` is doing, which is to provide a config file, or ncurses `make menueconfig` similar to how to configure the Linux kernel. The user will do `make meuconfig`, choose the target platform, processor type, and that is all. `Makefile will take care of everything else. 

@zoq I matched the version here to be the same as the one in `download_dependencies.sh` script in `board directory.
These are the default values, otherwise the user can set a new path for these deps

Reply:
Right, I see that we need to use the system C++ headers specific to the target host, so I agree that a user will need to tell us where those are.  (I guess that is part of the toolchain file.)  But we should still be able to use, e.g., cereal, Boost, and Armadillo headers that are installed on the system.  Maybe I overlooked something?

Reply:
Thanks for clarifying this for me today---I had not realized that if, e.g., `CEREAL_INCLUDE_DIR` is set to `/usr/include/` that this will translate into `-I/usr/include/`, which is a disaster for cross-compilation. :)  My thinking is that we can provide an auto-downloader for convenience, but also we should make it easy so that a user who knows what they are doing can already have set up all the dependencies in their cross-compilation environment (or elsewhere) and then can specify, e.g., `CEREAL_INCLUDE_DIR`, `ARMADILLO_INCLUDE_DIR`, etc. without problems. :+1: